app/controllers/spree/payments_controller.rb

```diff
@@ -14,16 +14,16 @@ def update
     end
 
     def find_payment
-      @payment = spree_current_user.payments.where(:id => params[:id]).first
+      @payment = spree_current_user.payments.where(:number => params[:id]).first
```

Comment:
@praveendhawan can't we use `find_by` here?
Also why are we using old hash syntax here?